### PR TITLE
Feature: Cancel search when pressing esc key

### LINKED ISSUE #9440: Feature: Cancel search when pressing esc key
### What's the Problem?

Currently if you are in the middle of a search and press escape, the app treats it like enter and continues the search. This goes against the escape button's supposed behavior.

### Requirements 

Pressing escape should cancel the search.

### Priorities

| Capability | Priority |
| :---------- | :------- |
| Pressing `esc` cancels search | Must |

### Files Version

Version: 2.3.0.0 

### Windows Version

10.0.19044.1766

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9440 

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/ViewModels/SearchBoxViewModel.cs [hunk 1]
```diff
@@ -71,6 +71,8 @@ public void SearchRegion_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextCh
 
 		public void SearchRegion_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs e)
 		{
+			if (e.ChosenSuggestion is null && string.IsNullOrWhiteSpace(e.QueryText))
+				return;
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SearchBoxViewModel.cs:75]:
```suggestion
				return;

```

File: src/Files.App/ViewModels/ToolbarViewModel.cs [hunk 2]
```diff
@@ -771,7 +771,7 @@ public void SearchRegion_LostFocus(object sender, RoutedEventArgs e)
 			CloseSearchBox();
 		}
 
-		private void SearchRegion_Escaped(object? sender, ISearchBox searchBox) => IsSearchBoxVisible = false;
+		private void SearchRegion_Escaped(object? sender, ISearchBox searchBox) => CloseSearchBox();
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/ToolbarViewModel.cs:774]: Is this still reachable?

Comment (ferrariofilippo) [reply src/Files.App/ViewModels/ToolbarViewModel.cs:774]: Yes, that method is subscribed to an event (Line # 381).
The `Escape` keyboard accelerator was duplicated (`AddressToolbar.xaml` and `SearchBox.xaml`)

**Review by yaichenbaum — APPROVED** [review]
Looks good, thank you!

**Pull request merged**